### PR TITLE
Add test-remove-memref-copy pass

### PR DESCRIPTION
**Pull request by JosseVanDelm**
Test pass that removes memref copy operations from the IR

### CODE REVIEW

File: tests/filecheck/transforms/realize-memref-casts.mlir
Comment (JosseVanDelm) [inline tests/filecheck/transforms/realize-memref-casts.mlir:1]: This was a weird thing we found while looking at the code

File: tests/filecheck/transforms/test-remove-memref-copy.mlir
Comment (JosseVanDelm) [inline tests/filecheck/transforms/test-remove-memref-copy.mlir:1]: This is pretty much what comes after "realize-memref-cast" @AntonLydike do you think I should put this inside that filecheck test instead?

File: tests/filecheck/transforms/test-remove-memref-copy.mlir [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+// CHECK-NEXT:       %14 = arith.extsi %arg4 : i8 to i32
+// CHECK-NEXT:       %15 = arith.subi %14, %arg6 : i32
+// CHECK-NEXT:       %16 = arith.muli %13, %15 : i32
+// CHECK-NEXT:       %17 = arith.addi %arg7, %16 : i32
+// CHECK-NEXT:       linalg.yield %17 : i32
+// CHECK-NEXT:     }
+// CHECK-NEXT:     func.return %arg2 : memref<?x128xi32, "L3">
+// CHECK-NEXT:   }
+// CHECK-NEXT: }
+
```
Comment (jorendumoulin) [inline tests/filecheck/transforms/test-remove-memref-copy.mlir:244]: unless these are testing something specific, these seem kind of overbodig

**Review by jorendumoulin — APPROVED** [review]
lmao

File: tests/filecheck/transforms/test-remove-memref-copy.mlir (hunk 1 above)
Comment (JosseVanDelm) [reply tests/filecheck/transforms/test-remove-memref-copy.mlir:244]: You are probably right :) will remove the overbodige code :)

**Pull request merged**